### DISCUSSION TITLE
.cache file with permissions

### DISCUSSION THREAD

**Discussion by stiebel** [ImageMagick/ImageMagick#3523] [Help]
Hi, I need a way to copy large images automatically from a /var/www/html directory to a user dir.
The web-created files are from www-data of course.
Unfortunately the .cache - file is with 600 permissions by default, but I need 644 to copy them as the requested user.
The .mpc file works fine.

How do I change the default permissions settings of the .cache files?

Greetings
Roland

**Comment by snibgo** [comment]
To change permissions of individual files, use "chmod".

Default permissions depend on attributes of parent directories. I don't know how those should be set to get what you want.

**Comment by stiebel** [comment]
> To change permissions of individual files, use "chmod".

I know that, and it is my plan B ;-)

> Default permissions depend on attributes of parent directories.

Normally yes, but the image.mpc has the right permissions, only the image.cache hat 600

So there must be some place to set this policy

**Comment by urban-warrior** [comment]
The image cache is a snapshot of your computer's memory that contains the image pixels.  For security reasons, only the owner, by default, can read or write the image cache.  The workflow owner can of course override this default behavior by changing the permissions of the cache file, e.g. chmod().

**Comment by stiebel** [comment]
Ok, Thank you.
It is not a great issue to chmod the file automatically, so I will just do it and will not bother anymore.
Greetings,
Roland
